tests/v2/validation/charts/backup_restore/backup_restore.go

```diff
@@ -0,0 +1,382 @@
+	clusterObject, _, err := provisioning.CreateProvisioningRKE1CustomCluster(client, &externalNodeProvider, testClusterConfig)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	provisioning.VerifyRKE1Cluster(t, client, testClusterConfig, clusterObject)
+
+	return clusterObject, testClusterConfig, nil
+}
+
+func createRKE2dsCluster(t *testing.T, client *rancher.Client) (*v1.SteveAPIObject, *clusters.ClusterConfig, error) {
+	provisioningConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, provisioningConfig)
+	nodeProviders := provisioningConfig.NodeProviders[0]
+	externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviders)
+
+	testClusterConfig := clusters.ConvertConfigToClusterConfig(provisioningConfig)
+
+	if provisioningConfig.RKE2KubernetesVersions == nil {
+		rke2Versions, err := kubernetesversions.ListRKE2AllVersions(client)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		provisioningConfig.RKE2KubernetesVersions = []string{rke2Versions[len(rke2Versions)-1]}
+	}
+
+	if provisioningConfig.CNIs == nil {
+		provisioningConfig.CNIs = []string{cniCalico}
+	}
+
+	nodeAndRoles := []provisioninginput.MachinePools{
+		provisioninginput.AllRolesMachinePool,
+	}
+	testClusterConfig.MachinePools = nodeAndRoles
+	testClusterConfig.KubernetesVersion = provisioningConfig.RKE2KubernetesVersions[0]
+	steveObject, err := provisioning.CreateProvisioningCustomCluster(client, &externalNodeProvider, testClusterConfig)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	provisioning.VerifyCluster(t, client, testClusterConfig, steveObject)
+
+	return steveObject, testClusterConfig, nil
+}
+
+func checkAWSS3Object(bucket string, key string) (bool, error) {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return false, err
+	}
+
+	s3Client := s3.NewFromConfig(sdkConfig)
+	_, err = s3Client.HeadObject(context.TODO(), &s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func deleteAWSObject(bucket string, key string) error {
+	sdkConfig, err := awsConfig.LoadDefaultConfig(context.TODO())
+	if err != nil {
+		return err
+	}
+	svc := s3.NewFromConfig(sdkConfig)
+
+	input := &s3.DeleteObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	}
+
+	_, err = svc.DeleteObject(context.TODO(), input)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+func verifyRancherResources(client *rancher.Client, curUserList []*management.User, curProjList []*management.Project, curRoleList []*management.RoleTemplate) (error, error, error) {
+	var userErr error
+	var projErr error
+	var roleErr error
+
+	logrus.Info("Verifying user resources...")
+	for i := range curUserList {
+		user, err := users.GetUserIDByName(client, curUserList[i].Name)
+		if err != nil {
+			userErr = err
+		}
+		if user == "" {
+			userErr = fmt.Errorf("user not found")
+		}
```

Comment:
the `GetUserIDByName` only returns an empty string, not an error so that's why I'm checking specifically for both in this one case. I discussed with @caliskanugur and rather than changing the `GetUserIDByName` function to return an error, I'll just check for an empty string since no user was found.